### PR TITLE
Refactor ChannelThreadExecutorGroup to ConnectionThreadExecutorGroup

### LINKED ISSUE #10184: Clear business logic with PostgreSQLBinaryStatementRegistry
The PostgreSQLBinaryStatementRegistry is different with MySQLBinaryStatementRegistry.

The scope of MySQLBinaryStatementRegistry is for proxy instance, all connections share same MySQLBinaryStatement map, key of binaryStatements is `statementID`, value is `SQL, parameters and parameter types`.

But the scope of PostgreSQLBinaryStatementRegistry is for connection instance, per connection hold independent PostgreSQLBinaryStatement map, key is `connectionID`, value is `ConnectionScopeBinaryStatementRegistry`, and `ConnectionScopeBinaryStatementRegistry` hold another map, key is `statementID`, value is `SQL, parameters and parameter types`.

Obviously, PostgreSQLBinaryStatementRegistry has one more layer than MySQLBinaryStatementRegistry.

Please clear the PostgreSQLBinaryStatementRegistry data structure. It is better to use same data structure with MySQLBinaryStatementRegistry if possible.

The PR #10185 is use status to control connection inactive. If using MySQL impl style, the PR #10185 maybe revert.

### PR DESCRIPTION
**Pull request by TeslaCN**
Fixes #10184.

Changes proposed in this pull request:
- Replace `ChannelId` with `ConnectionId`.
- Refactor `ChannelThreadExecutorGroup` to `ConnectionThreadExecutorGroup`.

### CODE REVIEW

File: shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-core/src/main/java/org/apache/shardingsphere/proxy/frontend/executor/ConnectionThreadExecutorGroup.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    public void unregisterAndAwaitTermination(final int connectionId) {
+        executorServices.computeIfPresent(connectionId, (unused, executorService) -> {
+            executorService.shutdown();
+            try {
+                executorService.awaitTermination(Long.MAX_VALUE, TimeUnit.MILLISECONDS);
+            } catch (final InterruptedException ignored) {
+                Thread.currentThread().interrupt();
+            }
+            return null;
+        });
```
Comment (terrymanu) [inline shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-core/src/main/java/org/apache/shardingsphere/proxy/frontend/executor/ConnectionThreadExecutorGroup.java:86]: We need to remove connectionId is unused.

**Review by terrymanu — CHANGES_REQUESTED** [review]

File: shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-core/src/main/java/org/apache/shardingsphere/proxy/frontend/executor/ConnectionThreadExecutorGroup.java (hunk 1 above)
Comment (TeslaCN) [reply shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-core/src/main/java/org/apache/shardingsphere/proxy/frontend/executor/ConnectionThreadExecutorGroup.java]: Indeed.

**Comment by TeslaCN** [comment]
I'm going to test this PR with BenchmarkSQL and sysbench before merging.

**Comment by TeslaCN** [comment]
I've tested it with sysbench and it works fine.
https://paste.ubuntu.com/p/yB49wZ7dzZ/

**Review by terrymanu — APPROVED** [review]

**Pull request merged**